library/std/src/net/ip_addr.rs

```diff
@@ -1875,7 +1875,7 @@ impl PartialEq<IpAddr> for Ipv6Addr {
     fn eq(&self, other: &IpAddr) -> bool {
         match other {
             IpAddr::V4(_) => false,
-            IpAddr::V6(v6) => self == v6,
+            IpAddr::V6(v6) => *self == *v6,
```

Comment:
Hmm, suggesting changes like this doesn't fill me with joy.

When in the compiler does the important difference between the two codegens happen?  Does it need to be in THIR->MIR lowering, or could a mir-opt notice these cases early enough to give the same benefits?

Reply:
Yeah, not suggesting this would be ideal. I wonder if the original problem is more with this happening in loops in particular. Having said that, I wanted to gauge how prevalent this is.

Reply:
Looking at the number of places where we are warning in rustc itself, I fear that 1) actually landing the suggestion should be gated on detecting the expression being part of a loop and 2) there might be real perf gains on fixing the underlining codegen optimization issue or cleaning up the ecosystem with the aid of this lint.